tools/run_tests/xds_k8s_test_driver/kubernetes-manifests/server.deployment.yaml

```diff
@@ -18,7 +18,9 @@ spec:
         app: ${deployment_name}
         owner: xds-k8s-interop-test
     spec:
+      % if service_account_name:
```

Comment:
This service account it OK, it's just a K8S internal service account.

Instead, do not add this annotation when Workload Identity is enabled.

https://github.com/grpc/grpc/blob/b6cc72f1295982472bba38e618c006cc943d2fc8/tools/run_tests/xds_k8s_test_driver/kubernetes-manifests/service-account.yaml#L9-L10

Reply:
In the client/server app, I tried to not explicitly create a service account for the given namespace. Is the K8s service account required for basic tests?

Reply:
Ah, OK. Technically K8S Service Account can be used for things other than Security and not directly dependent upon Workload Identity, like RBAC https://kubernetes.io/docs/reference/access-authn-authz/rbac/#service-account-permissions.

So with time this logic of not creating Service Account when Workload Identity disabled may change.
However, I'm OK if you want to keep it as if until then.

Reply:
Good to learn that. RBAC might be considered one of the security feature and tested in the security test suite. We are still testing the service account managing code. I would propose to keep it simple now, until the use case presents itself.